tools/hack/manage-mac-net-connect.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+MODE="${1:-setup}"  # setup | cleanup
+DOCKER_MAC_NET_CONNECT=${DOCKER_MAC_NET_CONNECT:-}
+HOMEBREW_GOPROXY=${HOMEBREW_GOPROXY:-}
+FLAG_FILE="/tmp/.docker-mac-net-connect-started"
+
+is_macos() {
+    [[ "$(uname -s)" == "Darwin" ]]
+}
+
+is_installed() {
+    [[ "$(brew list --formula | grep -Fx "docker-mac-net-connect")" == "docker-mac-net-connect" ]]
+}
+
+is_running() {
+    brew services info docker-mac-net-connect --json 2>/dev/null | \
+        jq -e '.[] | .user != null' >/dev/null
+}
+
+setup() {
+    is_macos || return
+
+    if [[ "$DOCKER_MAC_NET_CONNECT" == "false" ]]; then
+        return
+    fi
+
+    if is_installed && is_running; then
+        echo "docker-mac-net-connect is already installed and running."
+        return
+    fi
+
+    read -rp "Install and start docker-mac-net-connect? [y/N]: " input
+    case "$(echo "$input" | tr '[:upper:]' '[:lower:]')" in
+        y|yes) ;;
+        *) return ;;
+    esac
+
+    if ! is_installed; then
+        [ -n "$HOMEBREW_GOPROXY" ] && export HOMEBREW_GOPROXY="$HOMEBREW_GOPROXY"
+        brew install chipmk/tap/docker-mac-net-connect
+    fi
+
+    if ! is_running; then
+        sudo brew services start chipmk/tap/docker-mac-net-connect
```

Comment:
Need `root` permission to start; therefore, it seems impossible to make it fully uninteractive. At this point, it will ask for the root password.